### PR TITLE
Limit `requirements.txt` files updated by renovate

### PR DESCRIPTION
**Pull request by AlexWaygood**
## Summary

Following https://github.com/astral-sh/ruff/commit/50ff5c754421474b353f94edfdc1fed7a8c3b82b, renovate is now filing PRs to update all `requirements.txt` files in Ruff:
- https://github.com/astral-sh/ruff/pull/12829
- https://github.com/astral-sh/ruff/pull/12823

The `requirements.txt` file being updated in ^both of those PRs isn't really appropriate for renovate to update, as it's generated by a `uv pip compile` command, and renovate hasn't implemented support for uv yet.

It turns out that the reason that renovate is doing this is because the `fileMatch` settings is _additive_ -- when you specify a regex under a `fileMatch` key, renovate uses that pattern _in addition_ to its default. (Specifying the pattern doesn't _override_ renovate's default.) To actually exclude files from being updated by renovate, we have to use `ignorePaths`. We haven't come across this before now, because the other package managers we have enabled in our renovate configs all have somewhat conservative defaults that will only update files found at the repository root. Renovate's `pip_requirements` package manager, however, will search for `requirements.txt` files anywhere and everywhere.

## Test Plan

```
~/dev/ruff (renovate-config-fix) % npx --yes --package renovate -- renovate-config-validator --strict    
(node:31231) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 INFO: Validating .github/renovate.json5
 INFO: Config validated successfully
```

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -14,12 +14,24 @@
...
-    fileMatch: ["^docs/requirements.*\\.txt$"],
+    // The default for this package manager is to run on all requirements.txt files:
+    // https://docs.renovatebot.com/modules/manager/pip_requirements/#file-matching
+    // `fileMatch` doesn't work for excluding files; to exclude `requirements.txt` files
+    // outside the `doc/` directory, we instead have to use `ignorePaths` with a "negative glob pattern".
+    // See:
+    // - https://docs.renovatebot.com/modules/manager/#ignoring-files-that-match-the-default-filematch
+    // - https://docs.renovatebot.com/configuration-options/#ignorepaths
+    // - https://docs.renovatebot.com/string-pattern-matching/#negative-matching
+    ignorePaths: ["!docs/**"]
```
Comment (dhruvmanila) [inline .github/renovate.json5:30]: I haven't looked at the docs but do you think it's still reasonable to keep the `requirements.*\\.txt` part?

**Review by dhruvmanila — APPROVED** [review]
Thanks! This is useful information as well. We should also ignore Cargo updates for `fuzz` crate.

File: .github/renovate.json5 (hunk 1 above)
Comment (dhruvmanila) [reply .github/renovate.json5:30]: I guess the best way to verify this would be to merge and check the dependency dashboard issue?

Comment (AlexWaygood) [reply .github/renovate.json5:30]: > I haven't looked at the docs but do you think it's still reasonable to keep the `requirements.*\\.txt` part?

The double negative of "ignoring a negative glob pattern" is kind of melting my brain. But I _think_ that makes sense, yes -- thanks!

Comment (AlexWaygood) [reply .github/renovate.json5:30]: It's also so strange to me that `fileMatch` needs to be a regex whereas `ignorePaths` needs to be a glob

**Comment by AlexWaygood** [comment]
Looks like everything's now working as intended. After merging this PR:
- renovate autoclosed the two bad PRs it had previously opened
- The `docs/*requirements.txt` files dependencies are still listed as renovate-tracked dependencies in the dependency dashboard: https://github.com/astral-sh/ruff/issues/10578

**Pull request merged**